src/transformers/modeling_bart.py

```diff
@@ -951,6 +950,10 @@ def get_encoder(self):
     def get_output_embeddings(self):
         return _make_linear_from_emb(self.model.shared)  # make it on the fly
 
+    def _do_output_past(self, *args, **kwargs):
+        """ We should always use the cache in generate."""
+        return True
```

Comment:
That's good for me! I don't like the general `_do_output_past()` function in `modeling_utils.py` very much. So maybe switch to this kind of logic in general? As a default, `_do_output_past` is False and can be overwritten by the models themselves.  Ideally if we would have namedtuples we could just work with the names directly `a.k.a` `outputs.names`